workflow/controller/indexes/pod_index.go

```diff
@@ -0,0 +1,17 @@
+package indexes
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+func PodPhaseIndexFunc() cache.IndexFunc {
+	return func(obj interface{}) ([]string, error) {
```

Comment:
Minor - but I don think this needs to return a `cache.IndexFunc` when it can be one:

```golang
func PodPhaseIndexFunc() ([]string, error) {
                pod, ok := obj.(*corev1.Pod)
		if !ok {
			return nil, nil
		}
		return []string{string(pod.Status.Phase)}, nil
}
```